src/ShaderTools/Hlsl/Binding/Binder.Declarations.cs

```diff
@@ -267,7 +267,7 @@ private BoundNode BindFunctionDefinition(FunctionDefinitionSyntax declaration, S
                 Bind(declaration.Semantic, BindVariableQualifier);
 
             var functionBinder = (functionOwner != null && functionOwner.Kind == SymbolKind.Class)
```

Comment:
Yes, I planed to merge `StructSymbol` and `ClassOrStructSymbol` into one class `StructSymbol` and then in constructor use `StructTypeSyntax.IsClass` to pass correct `SymbolKind` to base class.